app/routes.py

```diff
@@ -220,44 +220,33 @@ def tower_settings(tower_id):
         print("form half_muffled is", form.half_muffled.data)
         tower.half_muffled = form.half_muffled.data
 
+        def emit_socket(name, data):
+            socketio.emit(name, data, broadcast=True, room=tower.tower_id)
+
         # ===== DEAL WITH WHEATLEY POTENTIALLY BEING ENABLED OR DISABLED =====
         wheatley_enabled = form.wheatley_enabled.data
         # Add this setting to the database's representation of the tower
         tower_db.wheatley_enabled = wheatley_enabled
         # If the enabledness of Wheatley has changed either way, broadcast that to the users of
         # that tower
         if tower.wheatley.enabled != wheatley_enabled:
-            socketio.emit('s_set_wheatley_enabledness', {'enabled': wheatley_enabled},
-                          broadcast=True, room=tower.tower_id)
+            emit_socket('s_set_wheatley_enabledness', {'enabled': wheatley_enabled})
             tower.wheatley.set_enabledness(wheatley_enabled)
             if wheatley_enabled:
                 # If Wheatley is being *enabled*, then update the setting in the tower and tell the
                 # clients that Wheatley has arrived
-                socketio.emit(
+                emit_socket(
                     's_user_entered',
                     {
                         'user_name': WHEATLEY_USER_NAME,
                         'user_id': WHEATLEY_USER_ID
                     },
-                    broadcast=True,
-                    include_self=True,
-                    room=tower.tower_id
                 )
-                socketio.emit('s_wheatley_setting', tower.wheatley.settings)
-                socketio.emit('s_wheatley_row_gen', tower.wheatley.row_gen)
+                emit_socket('s_wheatley_setting', tower.wheatley.settings)
+                emit_socket('s_wheatley_row_gen', tower.wheatley.row_gen)
             else:
-                # If Wheatley is being *disabled*, then update the setting in the tower and tell the
-                # clients that Wheatley has left
-                socketio.emit(
-                    's_user_left',
-                    {
-                        'user_name': WHEATLEY_USER_NAME,
-                        'user_id': WHEATLEY_USER_ID
-                    },
-                    broadcast=True,
-                    include_self=True,
-                    room=tower.tower_id
-                )
+                # If Wheatley is being *disabled*, tell the clients that Wheatley has left
+                emit_socket('s_user_left', {'user_id': WHEATLEY_USER_ID})
```

Comment:
I assume the `include_self` is safe to drop as there isn't a socketio caller, these are just initiated by the server?

As a separate point if there's load balancing, presumably the tower settings page can be served from a different server to the tower? So none of these would reach clients connected to that tower? That would be an existing bug that's probably quite hard to solve.

Reply:
> I assume the `include_self` is safe to drop as there isn't a socketio caller, these are just initiated by the server?

Ah yes I should fix this, but not at 1am....

> As a separate point if there's load balancing, presumably the tower settings page can be served from a different server to the tower? So none of these would reach clients connected to that tower? That would be an existing bug that's probably quite hard to solve.

That's a good point, but I don't think it's too much of an issue because it'll only affect half the towers and at that only when someone enables Wheatley with other people already in the tower, which surely can't be that common.

Reply:
Oh wow my brain was not working it's already fixed...